torch/optim/nadam.py

```diff
@@ -276,12 +285,13 @@ def _single_tensor_nadam(
     momentum_decay: float,
     eps: float,
     decoupled_weight_decay: bool,
+    maximize: bool,
     capturable: bool,
     differentiable: bool,
     has_complex: bool,
 ):
     for i, param in enumerate(params):
-        grad = grads[i]
+        grad = grads[i] if not maximize else -grads[i]
```

Comment:
this leads to materialization copy of grads, right?

Reply:
do you mean it’ll make a copy to compute the negation instead of just referencing?

if so, i think so, but it’s ok

Reply:
Yeah, maybe the negative bit functionality can be cool here :) Originally I was thinking that it could negate inplace... but maybe it's not a very widely used feature anyway...

Reply:
we don’t wanna modify the original grad in place!